aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -715,6 +722,119 @@ module aptos_framework::delegation_pool {
+            unlock_stake_events: account::new_event_handle<UnlockStakeEvent>(&stake_pool_signer),
+            withdraw_stake_events: account::new_event_handle<WithdrawStakeEvent>(&stake_pool_signer),
+            distribute_commission_events: account::new_event_handle<DistributeCommissionEvent>(&stake_pool_signer),
+        };
+
+        // Grant ownership of the existing stake to the participants of the destroyed staking contract.
+        // 1. `synchronize_delegation_pool` MUST NOT be called during this process because stake would be distributed as
+        // rewards instead of being individually allocated
+        // 2. partial governance voting remains disabled for simplicity
+
+        // allocate active stake to staker, any active commission has already been unlocked + allocated to operator
+        // no need to assert on minimum required stake as any next operation of delegator enforces it
+        buy_in_active_shares(&mut pool, staker_address, principal);
+
+        // allocate pending_inactive stake to staker, current operator, any previous operators still pending_inactive
+        while (pool_u64_iterable::shareholders_count(&distribution_pool) > 0) {
+            let recipients = pool_u64_iterable::shareholders(&distribution_pool);
+            let recipient = *vector::borrow(&recipients, 0);
+            let current_shares = pool_u64_iterable::shares(&distribution_pool, recipient);
+            let amount_to_distribute = pool_u64_iterable::redeem_shares(
+                &mut distribution_pool,
+                recipient,
+                current_shares
+            );
+
+            // no need to assert on minimum required stake as any next operation of delegator enforces it
+            buy_in_pending_inactive_shares(&mut pool, recipient, amount_to_distribute);
+        };
+        // destroy distribution pool of 0 remaining coins, any dust will be distributed as pending_inactive rewards
+        pool_u64_iterable::destroy_empty(distribution_pool);
+
+        // store `DelegationPool` on the resource account storing the `StakePool` of the destroyed staking contract
+        // `synchronize_delegation_pool` could not have been called as it acquires `DelegationPool` firstly stored here
+        move_to(&stake_pool_signer, pool);
+
+        // store ownership capability of delegation pool on staker
+        move_to(staker, DelegationPoolOwnership { pool_address });
+
+        // enable partial governance voting
+        if (features::partial_governance_voting_enabled(
+        ) && features::delegation_pool_partial_governance_voting_enabled()) {
+            // get staker's voter before being set to the resource account
+            let staker_delegated_voter = stake::get_delegated_voter(pool_address);
+
+            // partial governance voting could not have been enabled as the following would fail
+            enable_partial_governance_voting(pool_address);
```

Comment:
It looks like this function will be aborted if it is already enabled. It could be an issue, right?

Reply:
partial voting is not enabled at the beginning of `initialize_delegation_pool_from_staking_contract` because `delegation_pool::GovernanceRecords` could not have been published to the resource account within the `staking_contract` module. This comment is mostly for what's happening within the initialize function up to this point